src/adhocracy_core/adhocracy_core/authentication/__init__.py

```diff
@@ -90,6 +93,46 @@ def wrapped_view(context, request):
     return wrapped_view
 
 
+def is_header_password(request: IRequest) -> bool:
+    """Check if request provided the password in the Password header."""
+    return UserPasswordHeader in request.headers
+
+
+def get_header_password(request: IRequest) -> str:
+    """Return the password in the Password header."""
+    if is_header_password(request):
+        return request.headers[UserPasswordHeader]
+
+
+def validate_password_header(view: callable):
+    """Decorator vor :term:`view` to check if the password header may be set.
+
+    :raise pyramid.httpexceptions.HTTPBadRequest: if password is invalid.
+    """
+    def wrapped_view(context, request):
+        password_is_set = is_header_password(request)
+        error = None
+        if password_is_set:
+            content = request.registry.content
+            user = request.user
+            password = get_header_password(request)
+            is_valid = user.is_password_valid(request.registry, password)
+            if not is_valid:
+                error = error_entry('header',
+                                    UserPasswordHeader,
+                                    'Invalid password')
+            is_required = content.is_password_required(context, request)
+            if not is_required:
+                error = error_entry('header',
+                                    UserPasswordHeader,
+                                    'Password not required')
+        if error:
+            request.errors.append(error)
+            raise HTTPBadRequest()
+        return view(context, request)
+    return wrapped_view
```

Comment:
Correct me if I am wrong, but as far as I can see this is missing the most important error case: `is_required and not password_is_set`